### ISSUE TITLE
Exporting Layout Projection Element using Id instead of Identity

### ISSUE THREAD

**Issue by humanoidcreature** [OrchardCMS/Orchard#7050]
When a Layouts Projection Element gets exported in the `<BaseElementState>` tag, it uses the Content Item's Id instead of the Identity. Because of this you can't use it in a setup recipe, because you can't tell at that point when the query gets created in the setup process what Id will it get and it's 99% that it gets different Id what's in the exported Layout Projection Element's `<BaseElementState>` .

**Comment by sfmskywalker** [comment]
Although the Id value of the referenced content item is part of the `<BaseElementState>, the import process does not actually use that value. If you look at the `OnExporting` method of `ProjectionElementDriver`, you'll notice that it exports the identity value of the referenced, and overrides any stores Id value upon import with the correct value.

The import/export should therefore work, but perhaps you found a bug? Is the import/export/recipe thing failing?

**Comment by humanoidcreature** [comment]
```
    <Element ElementTypeName="NotificationProjection" BaseElementTypeName="Orchard.Layouts.Elements.Projection" ElementDisplayName="My Notification" ElementDescription="Projection element for notifications." ElementCategory="Notifications">
      <BaseElementState><![CDATA[QueryLayoutId=72%3b-1&ItemsToDisplay=0&Skip=0&MaxItems=20&PagerSuffix=]]></BaseElementState>
    </Element>
  </CustomElements>
```
Basically the import/export is working, but as you can see inside the `<BaseElementState>` there's a QueryLayoutId=[Content Item Id] part, but when you setup a site with a lot of Content Item, you can't be sure that the query's Id will be the same at that point when the setup's process reach this item.

**Comment by sfmskywalker** [comment]
Yes, but that value is not being used during import, as its overridden in the OnImporting method.
Although I agree that exporting the QueryLayoutId value is useless, this is an artifact of how element state is stored - when exporting, the entire state is dumped into the output.

Just to clarify your second statement: observation: is it taking the first query, or the first layout of that query? In either case, that is a bug. Which Orchard version are you looking at?

**Comment by sebastienros** [comment]
Marking it as discussion until someone confirm there is a bug.
